### ISSUE TITLE
VLLM bug fix

### ISSUE THREAD

**Issue by zywang0104** [lcqysl/FrameThinker_open#5]
# Refer to: https://github.com/vllm-project/vllm/issues/1728
    # and https://github.com/vllm-project/vllm/issues/15976
    def process_bad_tokens(token_ids, logits, exclude_token_ids=[]):
        for token_id in exclude_token_ids:
            logits[token_id] = -9999.999
        return logits

    # NOTE: tmp for visual agent!
    exclude_func = partial(process_bad_tokens, exclude_token_ids=[
        151643,    # <|endoftext|>
        151644,    # <|im_start|>
        151645,    # <|im_end|>
    ])
    agent_sampling_params.logits_processors = [exclude_func]
    agent_sampling_params.bad_words = ["<|endoftext|>", "<|im_start|>", "<|im_end|>"]

Hi, may I ask for some advice on this issue.. When using vLLM v1, it shows that custom modification of logits_processors is not allowed. However, after setting export VLLM_USE_V1=0, the training runs into CUDA OOM. Have you encountered this problem? Thanks in advance!
